include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+    DeleteFile(fileName.c_str());
+    return true;
+}
+#else
+inline bool _checkWritability(std::string path)
+{
+    struct stat d_stat;
+    if (stat(path.c_str(), &d_stat) < 0)
+    {
+        // try to make dir
+        std::string makeDir = "mkdir -p " + path;
+        if (system(makeDir.c_str()))
+            return false; // could not create home dir
+
+        if (stat(path.c_str(), &d_stat) < 0) // repeat stat
+            return false;
+    }
+
+    if (!(d_stat.st_mode & S_IWUSR))
+        return false; // dir not writable
+
+    return true;
+}
+#endif
+
+// ----------------------------------------------------------------------------
+// Function _getFileTimeDiff()
+// ----------------------------------------------------------------------------
+#if defined(PLATFORM_WINDOWS)
+inline double _getFileTimeDiff(std::string version_file)
+{
+    FILETIME curr;
+    GetSystemTimeAsFileTime(&curr);
+
+    HANDLE hFile;
+    FILETIME hFileTime;
+    hFile = CreateFile(version_file.c_str(), GENERIC_READ, FILE_SHARE_READ,  NULL,  OPEN_EXISTING,  FILE_ATTRIBUTE_NORMAL, NULL);
+
+    if(hFile == INVALID_HANDLE_VALUE)
+        return -1;
+    
+    if(!GetFileTime(hFile, NULL, NULL, &hFileTime))
+        return -1;
+    
+    ULARGE_INTEGER ul_curr;
+    ULARGE_INTEGER ul_file;
+    ul_curr.LowPart  = curr.dwLowDateTime;
+    ul_curr.HighPart = curr.dwHighDateTime;
+    ul_file.LowPart  = hFileTime.dwLowDateTime;
+    ul_file.HighPart = hFileTime.dwHighDateTime;
+
+    ULONGLONG diffInTicks = ul_curr.QuadPart - ul_file.QuadPart; // get time difference
+    LONGLONG diffInS = diffInTicks / 10000000;
+    CloseHandle(hFile);
+    return((double)diffInS);
+}
+#else
+inline double _getFileTimeDiff(std::string version_file)
+{
+    time_t curr;
+    time(&curr); // get current time
+
+    struct stat t_stat;
+    if (stat(version_file.c_str(), &t_stat) < 0)  // file does not exist
+        return -1;
+    else
+        return (difftime(curr, t_stat.st_mtime)); // returns time in seconds
+}
+#endif
+
+// ----------------------------------------------------------------------------
+// Function _getNumbersFromString()
+// ----------------------------------------------------------------------------
+inline void _getNumbersFromString(seqan::String<int> & numbers, std::string const & str)
+{
+    std::string number;
+    std::istringstream iss(str);
+    while (std::getline(iss, number, '.'))
+    {
+        if (!number.empty())
+        {
+            seqan::appendValue(numbers, atoi(seqan::toCString(number)));
+        }
+    }
+}
+
+// ----------------------------------------------------------------------------
+// Function _getVersionNumbers()
+// ----------------------------------------------------------------------------
+inline bool _readVersionNumbers(seqan::String<int> & version_numbers, std::string const & version_file)
+{
+    std::ifstream myfile;
+    myfile.open(version_file.c_str());
+    std::string line;
+    if (myfile.is_open())
```

Comment:
Maybe:
```cpp
if (!myfile.is_open())
    return false;

bool state = false;
// check regex and set state to true if successful ...
// otherwise print comment if unregistered app.
myfile.close();
return state;
```